src/kriscv/term_builder.py

```diff
@@ -267,8 +267,27 @@ def sb_bytes_cons(bs: KInner, rest_ef: KInner) -> KInner:
     return KApply('SparseBytes:BytesCons', bs, rest_ef)
 
 
-def sparse_bytes(data: dict[int, bytes]) -> KInner:
+def length_bytes(var: str) -> KInner:
+    return KApply('lengthBytes(_)_BYTES-HOOKED_Int_Bytes', [KVariable(var, 'Bytes')])
+
+
+def add_bytes(bytes1: KInner, bytes2: KInner) -> KInner:
+    return KApply('_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes', bytes1, bytes2)
+
+
+def sparse_bytes(data: dict[int, bytes], symdata: dict[int, tuple[int, str]] | None = None) -> KInner:
```

Comment:
I wonder if this is the right input representation. Maybe something like a

```python3
Sequence[bytes | int | tuple[KInner, int]]
```

or a similar class hierarchy would be more appropriate, since with the introduction of `KInner` sections we can no longer tell what single byte value is at a given address. This probably also removes much of the complexity of constructing the `KInner`.